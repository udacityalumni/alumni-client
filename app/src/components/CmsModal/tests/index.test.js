import CmsModal from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import mockTags, { selectedTags } from './mocks';

describe('<CmsModal />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <CmsModal
        isShowing
        onClose={e => e}
        spotlighted
        onToggleSpotlight={e => e}
        onSetStatus={e => e}
        status={0}
        onSave={e => e}
        canSubmit
        tags={mockTags}
        onChangeValue={e => e}
        selectedTags={selectedTags}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
