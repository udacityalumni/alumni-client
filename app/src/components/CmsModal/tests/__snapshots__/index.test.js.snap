exports[`<CmsModal /> should render with default props 1`] = `
<Layer
  align="center"
  closer={true}
  hidden={false}
  onClose={[Function onClose]}>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="medium"
    responsive={true}
    tag="div">
    <Heading
      align="center"
      tag="h1">
      Article Settings
    </Heading>
  </Box>
  <Form
    compact={false}
    fill={false}
    pad="large">
    <FormFields>
      <FormField
        label="Settings"
        size="medium"
        strong={false}>
        <CheckBox
          checked={true}
          id="spotlight-toggle"
          label="Spotlight"
          name="spotlight"
          onChange={[Function onToggleSpotlight]}
          toggle={true} />
      </FormField>
      <FormField
        label="Status"
        size="medium"
        strong={false}>
        <Select
          label="Status"
          onChange={[Function onChange]}
          options={
            Array [
              Object {
                "label": "Draft",
                "value": 0
              },
              Object {
                "label": "Published",
                "value": 1
              },
              Object {
                "label": "Archived",
                "value": 2
              }
            ]
          }
          value="Draft" />
      </FormField>
      <FormField
        label="Tags"
        size="medium"
        strong={false}>
        <undefined
          onChangeValue={[Function onChangeValue]}
          selectedTags={
            Array [
              Object {
                "id": "1",
                "slug": "software-engineering",
                "tag": "Software Engineering"
              }
            ]
          }
          tags={
            Array [
              Object {
                "id": "1",
                "slug": "software-engineering",
                "tag": "Software Engineering"
              },
              Object {
                "id": "2",
                "slug": "open-source",
                "tag": "Open Source"
              },
              Object {
                "id": "3",
                "slug": "mobile",
                "tag": "Mobile"
              },
              Object {
                "id": "4",
                "slug": "web",
                "tag": "Web"
              },
              Object {
                "id": "5",
                "slug": "machine-learning",
                "tag": "Machine Learning"
              },
              Object {
                "id": "6",
                "slug": "data-analysis",
                "tag": "Data Analysis"
              },
              Object {
                "id": "7",
                "slug": "javascript",
                "tag": "JavaScript"
              },
              Object {
                "id": "8",
                "slug": "react",
                "tag": "React"
              },
              Object {
                "id": "9",
                "slug": "graphql",
                "tag": "GraphQL"
              },
              Object {
                "id": "10",
                "slug": "ruby-on-rails",
                "tag": "Ruby on Rails"
              },
              Object {
                "id": "11",
                "slug": "salary-negotiation",
                "tag": "Salary Negotiation"
              }
            ]
          } />
      </FormField>
    </FormFields>
    <Footer
      align="center"
      direction="row"
      justify="center"
      pad="medium"
      responsive={false}>
      <Button
        label="Publish"
        onClick={[Function onSave]}
        type="button" />
    </Footer>
  </Form>
</Layer>
`;
